components/camel-aws-s3/src/main/java/org/apache/camel/component/aws/s3/S3Consumer.java

```diff
@@ -66,9 +67,12 @@ protected int poll() throws Exception {
 
         String fileName = getConfiguration().getFileName();
         String bucketName = getConfiguration().getBucketName();
+        String doneFileName = getConfiguration().getDoneFileName();
         Queue<Exchange> exchanges;
 
-        if (fileName != null) {
+        if (shouldSkipCauseDoneFileIsConfiguredButMissing(bucketName, doneFileName)) {
```

Comment:
Naming is hard but this name is really long :-D

Reply:
What do you think about the following:

```java
if (!doneFileCheckPasses(bucketName, doneFileName)) {
```

Or I was thinking about something like:

```java
if (doneFileCheckPasses(bucketName, doneFileName)) {
    // keep everything that was here beforehand
} else {
    exchanges = new LinkedList<>();
}
```


but that would mean a lot of whitespace changes and I don't want to create unnecessary changes.